src/components/code-diff-editor/code-diff-editor.tsx

```diff
@@ -0,0 +1,223 @@
+import { Component, Host, Prop, Watch, h } from "@stencil/core";
+import monaco, {
+  configureMonacoYaml
+} from "../../common/monaco/output/monaco.js";
+import { CodeDiffEditorOptions } from "./code-diff-editor-types.js";
+
+let autoId = 0;
+
+@Component({
+  shadow: false,
+  styleUrl: "code-diff-editor.scss",
+  tag: "ch-code-diff-editor"
+})
+export class ChCodeDiffEditor {
+  #monacoDiffEditorInstance!: monaco.editor.IStandaloneDiffEditor;
+  #resizeObserver: ResizeObserver | undefined;
+
+  #editorId: string = "";
+
+  // Refs
+  #monacoRef: HTMLDivElement;
+  #absoluteContentRef: HTMLDivElement;
+
+  /**
+   * Specifies the language of the auto created model in the editor.
+   */
+  @Prop() readonly language!: string;
+  @Watch("language")
+  languageChanged(newLanguage: string) {
+    this.#monacoDiffEditorInstance.setModel({
+      original: monaco.editor.createModel(this.value, newLanguage),
+      modified: monaco.editor.createModel(this.modifiedValue, newLanguage)
+    });
+  }
+
+  /**
+   * Specifies the theme to be used for rendering.
+   */
+  @Prop() readonly theme: string = "vs";
+  @Watch("theme")
+  themeChanged(newTheme: string) {
+    this.#monacoDiffEditorInstance.updateOptions({
+      theme: newTheme
+    });
+  }
+
+  /**
+   * Specifies the editor options.
+   */
+  @Prop() readonly options: CodeDiffEditorOptions = {
+    theme: this.theme,
+    automaticLayout: true,
+    mouseWheelScrollSensitivity: 4,
+    mouseWheelZoom: true,
+    enableSplitViewResizing: true,
+    renderSideBySide: true
+  };
+
+  /**
+   * Specifies if the editor should be readonly.
+   */
+  @Prop({ attribute: "readonly" }) readonly readonly: boolean = true;
+  @Watch("readonly")
+  readonlyChanged(newReadonly: boolean) {
+    if (this.#monacoDiffEditorInstance) {
+      this.#monacoDiffEditorInstance.updateOptions({
+        originalEditable: !newReadonly ?? true,
+        readOnly: newReadonly ?? false
+      });
```

Comment:
This implementation should match the logic implemented in the `#setupDiffEditor` method.